src/games/gameitem.py

```diff
@@ -285,9 +290,13 @@ def update(self, message, old_client=None):
         # Spawn announcers: IF we had a gamestate change, show replay and hosting announcements
         if oldstate != self.state:
             if self.state == "playing":  # The delay is there because we have a 5 minutes delay in the livereplay server
-                QtCore.QTimer.singleShot(5*60000, self.announceReplay)
+                self.launched_at = message['launched_at']
+                if time.time() - self.launched_at > 5*60:  # most already running games at client start
+                    self.announceReplay()
+                else:
+                    QtCore.QTimer().singleShot(5*60100 - 1000*(time.time() - self.launched_at), self.announceReplay)
```

Comment:
Well, there's also the coop widget which is also interested in hosted games, so I feel having a dedicated class to keep track of games is a nice idea. I called the folder 'model', because I envision it containing the 'model' classes like Player, Game, Mod etc., and other folders are named more after a specific tab than a specific functionality.